test/check-application

```diff
@@ -15,37 +15,64 @@ import testlib
+        browser.wait_popup('display-language')
+        browser.set_val("#display-language select", "de-de")
+        browser.click("#display-language-select-button")
+        browser.expect_load()
+        # HACK: work around language switching in Chrome not working in current session (Cockpit issue #8160)
+        browser.reload(ignore_cache=True)
+        browser.wait_present("#content")
+
+        # Then the menu label (from manifest) should be translated
+        browser.wait_text("#host-apps a[href='/starter-kit']", "Bausatz")
+
+        browser.go("/starter-kit")
+        browser.enter_page("/starter-kit")
+        # And the page label (from js) should be translated
+        browser.wait_in_text(".container-fluid p", "Läuft auf")
+
+    def testChangingHostnameFileWatching(self):
+        # Given on a cockpit instance
+        browser = self.browser
+        machine = self.machine
+        initial_hostname = machine.execute("cat /etc/hostname").strip()
+
+        self.login_and_go("/starter-kit")
+        self.goToStarterKit(browser, machine)
+
+        # When I change the hostname file on the underlying machine
+        machine.execute("echo new_$(hostname) > /etc/hostname").strip()
+        new_hostname = machine.execute("cat /etc/hostname").strip()
+
+        # Then the hostname is also updated in the starter page
+        browser.wait_text(".container-fluid p", "Running on " + new_hostname)
+        assert new_hostname == "new_" + initial_hostname
+
+    def goToStarterKit(self, browser, machine):
         # verify expected heading
-        b.wait_present(".container-fluid h2")
-        b.wait_text(".container-fluid h2", "Starter Kit")
+        browser.wait_present(".container-fluid h2")
+        browser.wait_text(".container-fluid h2", "Starter Kit")
 
         # verify expected host name
-        hostname = m.execute("hostname").strip()
-        b.wait_present(".container-fluid p")
-        b.wait_text(".container-fluid p", "Running on " + hostname)
-
-        # change language to German
-        b.switch_to_top()
-        b.click("#content-user-name")
-        b.click(".display-language-menu a")
-        b.wait_popup('display-language')
-        b.set_val("#display-language select", "de-de")
-        b.click("#display-language-select-button")
-        b.expect_load()
-        # HACK: work around language switching in Chrome not working in current session (Cockpit issue #8160)
-        b.reload(ignore_cache=True)
-        b.wait_present("#content")
-        # menu label (from manifest) should be translated
-        b.wait_text("#host-apps a[href='/starter-kit']", "Bausatz")
-
-        b.go("/starter-kit")
-        b.enter_page("/starter-kit")
-        # page label (from js) should be translated
-        b.wait_in_text(".container-fluid p", "Läuft auf")
+        hostname = machine.execute("hostname").strip()
+        browser.wait_present(".container-fluid p")
+        browser.wait_text(".container-fluid p", "Running on " + hostname)
```

Comment:
This is not conceptually a part of "go to the starter kit page", it's an independent check.